### DISCUSSION TITLE
Container Registry Images not consuming storage

### DISCUSSION THREAD

**Discussion by JeremyGrieshop** [community/community#21973] [New to GitHub]
<p>I’ve pushed two private images to GitHub’s container registry, where my Free Org is supposed to have a 0.5 GB limit.  The total space, according to <code>docker image ls</code> is 1.7 GB and 1.3 GB.  My billing still shows virtually no storage used.</p>
<p>What’s going on?  In general, if the image uses layers that GitHub may already have cached, do those count against my storage, or not?</p>
<p>Thanks!</p>

**Comment by JeremyGrieshop** [comment]
<p>In case anyone else finds this and needs to know the answer - as of 9/1/2021, the Container Registry consumes no storage.  This won’t be permanent, though.</p>
<p>All Storage currently consumed in Billing accounts for packages and artifacts created by GitHub Actions.</p>

**Comment by wardboumans** [comment]
<p><a class="mention" href="/jeremygrieshop">@JeremyGrieshop</a> can you share where this is explained? Im recently noticed this ‘issue’ too. I want to avoid surprises.</p>

**Comment by JeremyGrieshop** [comment, marked as answer]
<p>Someone from GitHub support replied to me on 9/1/2021:</p>
<blockquote>
<p>Thanks for reaching out, I’m sorry for the confusion here! We are not currently charging for storage related to GitHub Package Registry. It was recently moved from beta to generally available, but billing has not been enabled yet. When it is, we will give at least 30 days notice so there will hopefully not be any surprises. Read more here:</p>
<p><a href="https://github.blog/2021-06-21-github-packages-container-registry-generally-available/" class="inline-onebox" rel="noopener nofollow ugc">GitHub Packages Container registry is generally available | The GitHub Blog</a></p>
</blockquote>

**Comment by wardboumans** [comment]
<p>Thank you! That clears it up. Guess we have to keep an eye on that notice.</p>

**Comment by jricciardi** [comment]
<p>Thank you for following up with that message from Support, <a class="mention" href="/jeremygrieshop">@JeremyGrieshop</a>! I’ve marked that as the ‘Solution’ here for any future users who find this post.</p>

**Comment by sksar** [comment]
Any updates as of July 2023?

**Reply by DASPRiD** [reply]
From what I can see in my usage report, it's still not being counted.

**Reply by DFMoller** [reply]
My images also are not counting towards package registry storage.

**Reply by TheAlexDev23** [reply]
Same here

**Reply by drmrbrewer** [reply]
This still seems to be the case?  I've only just started using GHCR and was confused to see no usage being reported on [this page](https://github.com/settings/billing) (under Packages at the bottom) despite now having private images in place.  But from [this page](https://docs.github.com/en/billing/managing-billing-for-your-products/about-billing-for-github-packages) it seems that these images *should* now be billable above the free quota?  I'm confused.  And a quota of 500 MB total seems ludicrously low... just a very basic image of mine is 440 MB.
